utils.py

```diff
@@ -59,16 +61,77 @@ def mutations_data(self):
 
         node_flag = ts.nodes_flags[mutations_node]
         position = ts.sites_position[ts.mutations_site]
-        df = pd.DataFrame({
-            "position": position,
-            "node": ts.mutations_node,
-            "time": mutations_time,
-        })
+
+        tables = self.ts.tables
+        assert np.all(
+            tables.mutations.derived_state_offset == np.arange(ts.num_mutations + 1)
+        )
+        derived_state = tables.mutations.derived_state.view("S1").astype(str)
+
+        assert np.all(
+            tables.sites.ancestral_state_offset == np.arange(ts.num_sites + 1)
+        )
+        ancestral_state = tables.sites.ancestral_state.view("S1").astype(str)
+        del tables
+        inherited_state = ancestral_state[ts.mutations_site]
+        mutations_with_parent = ts.mutations_parent != -1
+
+        parent = ts.mutations_parent[mutations_with_parent]
+        assert np.all(parent >= 0)
+        inherited_state[mutations_with_parent] = derived_state[parent]
+        self.mutations_derived_state = derived_state
+        self.mutations_inherited_state = inherited_state
+
+        self.mutations_position = ts.sites_position[ts.mutations_site].astype(
+            int
+        )
+        N = ts.num_mutations
+        mutations_num_descendants = np.zeros(N, dtype=int)
+        mutations_num_inheritors = np.zeros(N, dtype=int)
+        mutations_num_parents = np.zeros(N, dtype=int)
+
+        tree = ts.first()
+        iterator = tqdm.tqdm(np.arange(N), desc="Classifying mutations")
```

Comment:
Let's get rid of the progress bar for now, it'll complicate things quite a bit. Can add later if we want it.